alpha/apps/kaltura/lib/storage/urlTokenizers/kCdnVideoUrlTokenizer.php

```diff
@@ -0,0 +1,43 @@
+<?php
+class kCdnVideoUrlTokenizer extends kUrlTokenizer
+{
+	/**
+	 * @param string $url
+	 * @param string $urlPrefix
+	 * @return string
+	 */
+	public function tokenizeSingleUrl($url, $urlPrefix = null)
+	{
+		return $this->tokenizeUrl($url);
+	}
+	/**
+	 * @param $url
+	 * @return string
+	 */
+	private function tokenizeUrl($url)
+	{
+		$expiryTime = time()+$this->getWindow();
+		$baseUrl = parse_url($url);
+		$hashData = $this->getKey().":$expiryTime:$baseUrl[path]";
+		$token = $this->getTokenByData($hashData);
+
+		if (strpos($url, '?') !== false)
+			$s = '&';
+		else
+			$s = '?';
+
+		return $url.$s.'md5='.$token.'&e='.$expiryTime;
```

Comment:
@TalDubovKaltura it may be more readable here to use double quotes instead of the multiple concats.

Reply:
Done